### PR TITLE
Add support for debug bus mastering

### PR DESCRIPTION
**Pull request by timsifive**
Tested with all the gdbserver tests, and an OpenOCD that knows how to use this feature.

### CODE REVIEW

File: riscv/debug_module.cc [hunk 1]
```diff
@@ -78,6 +68,14 @@ void debug_module_t::reset()
   abstractcs.progbufsize = progbufsize;
 
   abstractauto = {0};
+
+  sbcs = {0};
```
Comment (mwachs5) [inline riscv/debug_module.cc:73]: Is system bus access a hard-coded feature? As well as the access types?

Comment (timsifive) [reply riscv/debug_module.cc:73]: Yes.

Comment (mwachs5) [reply riscv/debug_module.cc:73]: Can we make it optional with a command line arg? S.t. SW can be tested against one that doesn't have system bus?

Comment (timsifive) [reply riscv/debug_module.cc:73]: Sure. Do you have suggestions for an option name? I'm having a hard time thinking of something concise but descriptive. Maybe --debug-sba? What do you want the default to be?

Currently in my testing I rely on OpenOCD to prefer the program buffer if it exists.

Comment (mwachs5) [reply riscv/debug_module.cc:73]: I think --debug-sba=<N> , where N is the maximum size it supports and the default is 0? (I don't think it makes sense to support some size and not support something smaller than that).

Comment (timsifive) [reply riscv/debug_module.cc:73]: Done.

**Comment by timsifive** [comment]
Can I merge this?

**Review by palmer-dabbelt — APPROVED** [review]

**Pull request merged**